File: typescript/difference-of-squares/difference-of-squares.ts
export class Squares {
  sumOfSquares_: number
  squareOfSum_: number
  difference_: number

  constructor(n: number) {
    this.squareOfSum_ = Math.pow(n * (n + 1) / 2, 2)
    this.sumOfSquares_ = (n * (n + 1) * (2 * n + 1)) / 6
    this.difference_ = Math.abs(this.sumOfSquares_ - this.squareOfSum_)
  }

  get sumOfSquares(): number {
    return this.sumOfSquares_;
  }

  get squareOfSum(): number {
    return this.squareOfSum_;
  }

  get difference(): number {
    return this.difference_;
  }
}

File: kotlin/difference-of-squares/src/main/kotlin/DifferenceOfSquares.kt
data class Squares(val n: Int) {
    fun sumOfSquares() = n * (n + 1) * (2 * n + 1) / 6

    fun squareOfSum() = (n * (n + 1) / 2).let { it * it }

    fun difference() = squareOfSum() - sumOfSquares()
}
